packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -49,6 +49,48 @@ export type MenuItemProps = {
   style?: React.CSSProperties;
 };
 
+type BasePlacement = 'top' | 'bottom' | 'left' | 'right';
+type VariationPlacement =
+  | 'top-start'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'right-start'
+  | 'right-end'
+  | 'left-start'
+  | 'left-end';
+type AutoPlacement = 'auto' | 'auto-start' | 'auto-end';
+type Placement = AutoPlacement | BasePlacement | VariationPlacement;
+
+type ClickAwayMouseEventHandler =
+  | 'onClick'
+  | 'onMouseDown'
+  | 'onMouseUp'
+  | 'onPointerDown'
+  | 'onPointerUp';
+type ClickAwayTouchEventHandler = 'onTouchStart' | 'onTouchEnd';
+
+export type PopperProps = {
+  open: boolean;
+  children?: React.ReactNode;
+  className?: string;
+  clickAwayTouchEvent?: false | ClickAwayTouchEventHandler;
+  clickAwayMouseEvent?: false | ClickAwayMouseEventHandler;
+  flip?: boolean;
+  focusTrap?: boolean;
+  focusTrapEnabled?: boolean;
+  onExited?: (node: HTMLElement | null) => void;
+  onClickAway?: (event: MouseEvent | TouchEvent) => void;
+  onDidMount?: () => void;
+  onDidUnmount?: () => void;
+  id?: string;
+  ref?: Ref;
+  target?: Element | null;
+  transition?: boolean;
```

Comment:
Do we need this in the base popper? It seems like a slot implementation detail to me.

Reply:
It felt necessary, do you see an alternative place where to put transition logic? Having it as a prop here avoids exposing MUI Grow and handling `TransitionProps` in the grid's core codebase.

Reply:
Oh, so we use poppers with and without transition in our codebase? If so, then it makes sense 👍🏻

Reply:
We do, the menus (column, density, export) have transitions and the panels (columns management, filters) don't. It could also make sense to expose different slots for different popper types (e.g. "menu" and "panel"), but for now this is the most concise diff so I'll go with that.